src/Forms/Form.php

```diff
@@ -1790,6 +1792,21 @@ public function removeExtraClass($class)
         return $this;
     }
 
+    public function __clone(): void
+    {
+        $this->fields = clone $this->fields;
+        $this->actions = clone $this->actions;
+        if ($this->validator) {
+            $this->setValidator(clone $this->validator);
+        }
+        foreach ($this->fields as $field) {
+            $field->setForm($this);
```

Comment:
Is this recursive? e.g. will field groups and composite fields ensure their descendants call $field->setForm() properly?

I'd kind of expect this to happen in the FieldList cloning you mentioned in the comment above, so I'm a little surprised to see it here

Reply:
`FieldList` does have some logic, `CompositeField` also has some logic. But the form being cloned is where the form should be set. Closing a `FieldList` doesn't mean the form is changing. The `FieldList` doesn't even keep track of the form, for that matter.